packages/app/src/Pages/settings/Profile.tsx

```diff
@@ -112,6 +125,103 @@ export default function ProfileSettings(props: ProfileSettingsProps) {
     }
   }
 
+  async function onNip05Change(e: React.ChangeEvent<HTMLInputElement>){
+    const Nip05Address = e.target.value;
+    setNip05(Nip05Address)
+    const Nip05AddressElements = Nip05Address?.split('@') ?? [];
+    if (Nip05Address.length === 0) {
+        setNip05AddressValid(false)
+        setInvalidNip05AddressMessage("")
+    }else if(Nip05AddressElements.length < 2){
+        setNip05AddressValid(false)
+        setInvalidNip05AddressMessage(formatMessage(messages.InvalidNip05Address))
+    }
+    else if(Nip05AddressElements.length === 2){
+      nip05NostrAddressVerification(Nip05AddressElements.pop(), Nip05AddressElements.pop());
+    }
+    else{
+      setNip05AddressValid(false)
+    }
+  }
+
+  async function onLimitCheck(val:string, field:string){
+    if(field === "username"){
+      setName(val);
+      if(val?.length >= MaxUsernameLength){
+        setUsernameValid(false)
+        setInvalidUsernameMessage(formatMessage(messages.UserNameLengthError))
+      }else{
+        setUsernameValid(true)
+        setInvalidUsernameMessage("")
+      }
+    }
+    else if(field === "about"){
+      setAbout(val);
+      if(val?.length >= MaxAboutLength){
+        setAboutValid(false)
+        setInvalidAboutMessage(formatMessage(messages.AboutLengthError))
+      }else{
+        setAboutValid(true)
+        setInvalidAboutMessage("")
+      }
+    }
+  }
+
+  async function nip05NostrAddressVerification(nip05Domain: string | undefined, nip05Name: string| undefined) {
+    if(!nip05Domain || !nip05Name)
+      return; 
+
+    try{
+      const res = await fetch(`https://${nip05Domain}/.well-known/nostr.json?name=${nip05Name}`);
+
+      if(res.status === 200){
+        const json = await res.json();
+        const hexKeyFromRemote = json['names'][`${nip05Name}`]
+
+        if (json['names'] && hexKeyFromRemote) {
+          if (id === hexKeyFromRemote) {
+            setNip05AddressValid(true);
+          }else{
+            setNip05AddressValid(false);
+            setInvalidNip05AddressMessage(formatMessage(messages.InvalidNip05Address))
+          }
+        }
+      }
+      else {
+          setNip05AddressValid(false)
+          setInvalidNip05AddressMessage(formatMessage(messages.ErrorValidatingNip05Address))
+      }
+    }catch(e){
+      setNip05AddressValid(false)
+      setInvalidNip05AddressMessage(formatMessage(messages.InvalidNip05Address))
+    }
+  }
+
+  async function onLud16Change(address:string){
```

Comment:
This would be better if it was triggered with `useEffect` and use `return debounce(() => fn)` to add a delay, this is what we do for search etc, you should be able to see this pattern used already